inventory/upcloud.py

```diff
@@ -232,11 +232,18 @@ def read_api_credentials(config):
 
     # setup API connection
     username, password = read_api_credentials(config)
-    default_timeout = os.getenv('UPCLOUD_API_TIMEOUT')
+    default_timeout = os.getenv('UPCLOUD_API_TIMEOUT') or config.get('upcloud', 'default_timeout')
     if not default_timeout:
-        default_timeout = config.get('upcloud', 'default_timeout') or None
+        default_timeout = None
+    else:
+        default_timeout = float(default_timeout)
     manager = upcloud_api.CloudManager(username, password, default_timeout)
 
+    # In case user doesn't use env variables, we might need them in modules
+    os.environ['UPCLOUD_API_USER'] = username
+    os.environ['UPCLOUD_API_PASSWD'] = password
+    os.environ['UPCLOUD_API_TIMEOUT'] = default_timeout
```

Comment:
Does this work? 🤔 Not familiar so not sure if it's possible to set env vars like this, sometimes they are readonly 🤔

Reply:
This could be one of those "Works on my machine" ™️ type of deals, so not 100% sure.

Reply:
If it works on your machine and doesn't crash then LGTM 😄